python/semantic_kernel/core_plugins/crew_ai/crew_ai_enterprise.py

```diff
@@ -0,0 +1,228 @@
+            client=client,
+            polling_interval=settings.polling_interval,
+        )
+
+    async def kickoff(
+        self,
+        inputs: dict[str, Any] | None = None,
+        task_webhook_url: str | None = None,
+        step_webhook_url: str | None = None,
+        crew_webhook_url: str | None = None,
+    ) -> str:
+        """Kickoff a new Crew AI task.
+
+        Args:
+            inputs (dict[str, Any], optional): The inputs for the task. Defaults to None.
+            task_webhook_url (str | None, optional): The webhook URL for task updates. Defaults to None.
+            step_webhook_url (str | None, optional): The webhook URL for step updates. Defaults to None.
+            crew_webhook_url (str | None, optional): The webhook URL for crew updates. Defaults to None.
+
+        Returns:
+            str: The ID of the kickoff response.
+        """
+        try:
+            kickoff_response = await self.client.kickoff(inputs, task_webhook_url, step_webhook_url, crew_webhook_url)
+            logger.info(f"CrewAI Crew kicked off with Id: {kickoff_response.kickoff_id}")
+            return kickoff_response.kickoff_id
+        except Exception as ex:
+            raise FunctionExecutionException("Failed to kickoff CrewAI Crew.") from ex
+
+    @kernel_function(description="Get the status of a Crew AI kickoff.")
+    async def get_crew_kickoff_status(self, kickoff_id: str) -> CrewAIStatusResponse:
+        """Get the status of a Crew AI task.
+
+        Args:
+            kickoff_id (str): The ID of the kickoff response.
+
+        Returns:
+            CrewAIStatusResponse: The status response of the task.
+        """
+        try:
+            status_response = await self.client.get_status(kickoff_id)
+            logger.info(f"CrewAI Crew status for kickoff Id: {kickoff_id} is {status_response.state}")
+            return status_response
+        except Exception as ex:
+            raise FunctionExecutionException(
+                f"Failed to get status of CrewAI Crew with kickoff Id: {kickoff_id}."
+            ) from ex
+
+    @kernel_function(description="Wait for the completion of a Crew AI kickoff.")
+    async def wait_for_crew_completion(self, kickoff_id: str) -> str:
+        """Wait for the completion of a Crew AI task.
+
+        Args:
+            kickoff_id (str): The ID of the kickoff response.
+
+        Returns:
+            str: The result of the task.
+
+        Raises:
+            FunctionExecutionException: If the task fails or an error occurs while waiting for completion.
+        """
+        try:
+            status_response = None
+            status = CrewAIEnterpriseKickoffState.Pending
+            while status not in [
+                CrewAIEnterpriseKickoffState.Failed,
+                CrewAIEnterpriseKickoffState.Failure,
+                CrewAIEnterpriseKickoffState.Success,
+                CrewAIEnterpriseKickoffState.Not_Found,
+            ]:
+                logger.info(
```

Comment:
maybe turn this to Debug, since it might take a while and lead to a whole bunch of the same logs